File: app/src/main/java/com/liadpaz/amp/adapters/MainViewPagerAdapter.java
package com.liadpaz.amp.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.liadpaz.amp.fragments.AlbumListFragment;
import com.liadpaz.amp.fragments.ArtistListFragment;
import com.liadpaz.amp.fragments.PlaylistsFragment;
import com.liadpaz.amp.fragments.SongsListFragment;
import com.liadpaz.amp.livedatautils.SongsUtil;

public class MainViewPagerAdapter extends FragmentStateAdapter {

    public MainViewPagerAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0: {
                return SongsListFragment.newInstance(SongsUtil.getSongs());
            }

            case 1: {
                return PlaylistsFragment.newInstance();
            }

            case 2: {
                return ArtistListFragment.newInstance();
            }

            case 3: {
                return AlbumListFragment.newInstance();
            }
        }
        throw new IndexOutOfBoundsException("Max 4 pages!");
    }

    @Override
    public int getItemCount() {
        return 4;
    }
}

File: app/src/main/java/com/liadpaz/amp/adapters/MainViewPagerAdapter.kt
package com.liadpaz.amp.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.liadpaz.amp.fragments.AlbumListFragment
import com.liadpaz.amp.fragments.ArtistListFragment
import com.liadpaz.amp.fragments.PlaylistsFragment
import com.liadpaz.amp.fragments.SongsListFragment.Companion.newInstance
import com.liadpaz.amp.livedatautils.SongsUtil

class MainViewPagerAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) : FragmentStateAdapter(fragmentManager, lifecycle) {
    override fun createFragment(position: Int): Fragment =
        when (position) {
            0 -> newInstance(SongsUtil.getSongs())
            1 -> PlaylistsFragment.newInstance()
            2 -> ArtistListFragment.newInstance()
            3 -> AlbumListFragment.newInstance()
            else -> throw IndexOutOfBoundsException("Max 4 pages!")
        }

    override fun getItemCount(): Int = 4
}